### DISCUSSION TITLE
Can we show Search result sorted by Main site then Hub sites and then rest of sites in the tenant

### DISCUSSION THREAD

**Discussion by vishalgoyal2472** [microsoft-search/pnp-modern-search#4063] [General]
can we achieve below?

if we search in search box result should be sorted as below

1. Main site where we implemented this pnp search
2. Hub site of main site
3. rest of sites in tenant

**Comment by kasperbolarsen** [comment]
I would think you can accomplish that by tweaking the ranking . Search for XRANK and there are a number of guides available

**Reply by vishalgoyal2472** [reply]
can we set rank for each site or for specific pnp result webpart we need to set the rank.

**Reply by vishalgoyal2472** [reply]
can we do any setting at pnp search webpart level so it will not affect entire tenant level search.

**Reply by kasperbolarsen** [reply]
Absolutely,  you can tweak the ranking in the search query by using XRANK,  and then only the results in this particular search web part is affected

**Reply by vishalgoyal2472** [reply]
can you please share any example pls?

**Reply by vishalgoyal2472** [reply]
> I would think you can accomplish that by tweaking the ranking . Search for XRANK and there are a number of guides available

I am trying for below results order

1. result from Site1 from domain1
2. result from site2 from domain 1
3. result from all sites from domain1
4. result from all sites from domain2

In query template I have put sites in below order, for first site I am getting correct result but after that I am getting random results.

```
{searchTerms} 
XRANK(cb=4000) path:https://doamin1.sharepoint.com/sites/site1
XRANK(cb=300) path:https://domain1.sharepoint.com/sites/site2
XRANK(cb=20) path:https://domain1.sharepoint.com/
XRANK(cb=1) path:https://domain.sharepoint.com/
```

**Reply by kasperbolarsen** [reply]
Perhaps the approach on this blog post can help,  https://www.techmikael.com/2016/12/tip-for-writing-multiple-xranks-or.html?m=1

**Reply by vishalgoyal2472** [reply]
If I need to use all hub sites linked with parent site, do I need to put all site URLs manually in search query or do we have any syntax for it?

**Reply by vishalgoyal2472** [reply]
I am trying to execute below query which is sorting the results correctly but there is one issue while I add XRANK to any site

Query1 and Query2 results are mismatching, Query1 showing more count and it seems its pulling all results from tenant.

 ### **Query1**
> {searchTerms} 
> XRANK(cb=4000) path:https://doamin1.sharepoint.com/sites/site1 
> OR
> XRANK(cb=300) path:https://domain1.sharepoint.com/sites/site2
 

###  **Query2**
> {searchTerms} 
> path:https://doamin1.sharepoint.com/sites/site1
>OR
> path:https://domain1.sharepoint.com/sites/site2

**Reply by wobba** [reply]
Do not use the cb parameter as you have no control of the rank score range. Use **nb**.

**Reply by vishalgoyal2472** [reply]
we used nb also but it did not work.

**Reply by vishalgoyal2472** [reply]
> I am trying to execute below query which is sorting the results correctly but there is one issue while I add XRANK to any site
> 
> Query1 and Query2 results are mismatching, Query1 showing more count and it seems its pulling all results from tenant.
> 
> ### **Query1**
> > {searchTerms}
> > XRANK(cb=4000) path:https://doamin1.sharepoint.com/sites/site1
> > OR
> > XRANK(cb=300) path:https://domain1.sharepoint.com/sites/site2
> 
> ### **Query2**
> > {searchTerms}
> > path:https://doamin1.sharepoint.com/sites/site1
> > OR
> > path:https://domain1.sharepoint.com/sites/site2

can anyone please respond on this or tried this scenario?

**Reply by wobba** [reply]
`(({searchTerms}) XRANK(nb=5) path:https://x.sharepoint.com/sites/site1) XRANK(nb=2) path:https://x.sharepoint.com/sites/site2`


Be sure to add () to keep any boolean logic you want achieved. I added them where I thought you needed them.
